core/lib/server/pub.ts

```diff
@@ -1059,14 +1068,20 @@ type MaybePubChildren<Options extends GetPubsWithRelatedValuesAndChildrenOptions
  * Only add the `stage` if the `withStage` option has not been set to `false
  */
 type MaybePubStage<Options extends GetPubsWithRelatedValuesAndChildrenOptions> =
-	Options["withStage"] extends true ? { stage: Stages } : { stage?: never };
+	Options["withStage"] extends true ? { stage?: Stages } : { stage?: never };
```

Comment:
because a pub might not belong to a stage

Reply:
a true! but then i think i should be `Stages | null` right?